server/src/main/java/org/opensearch/search/fetch/subphase/MatchedQueriesPhase.java

```diff
@@ -71,35 +69,41 @@ public FetchSubPhaseProcessor getProcessor(FetchContext context) throws IOExcept
         for (Map.Entry<String, Query> entry : namedQueries.entrySet()) {
             weights.put(
                 entry.getKey(),
-                context.searcher().createWeight(context.searcher().rewrite(entry.getValue()), ScoreMode.COMPLETE_NO_SCORES, 1)
+                context.searcher().createWeight(context.searcher().rewrite(entry.getValue()), ScoreMode.COMPLETE, 1)
```

Comment:
@msfroh would love to hear your opinion: I believe this change could have a visible impact on search (since always calculate scores now). If that is the case, we should be able to pass the state of the `include_named_queries_score` parameter through `FetchContext` and only calculate scores when asked explicitly. What do you think?

Reply:
Thats a good point. I'm also a little concerned about the changes in the processor logic itself. For example, we never check for a two-phase approximation for the scorer (which `Lucene.asSequentialAccessBits` does). Depending on the queries that we're using for matching, this could increase the matching cost. 

(Of course, most scorers that implement `twoPhaseIterator()` use the same implementation for `iterator()`. It looks like `SpanScorer` is an exception, though.)

I'm wondering if it might be worth splitting the `getProcessor` implementation for the case where we want scores?

@Dharin-shah -- thank you very much for implementing this! What approach do you prefer to make sure that the non-scoring behavior doesn't change (functionally and performance-wise) while making sure that scores are available when requested?

Reply:
okay can make some changes, will try and avoid interface breakage, but if it happens i guess its fine

Reply:
so before implementing @msfroh was wondering, can i just use FetchContext to determine if we want to return scores or not?

`  @Override
    public FetchSubPhaseProcessor getProcessor(FetchContext context) throws IOException {
        Map<String, Query> namedQueries = new HashMap<>();
        if (context.parsedQuery() != null) {
            namedQueries.putAll(context.parsedQuery().namedFilters());
        }
        if (context.parsedPostFilter() != null) {
            namedQueries.putAll(context.parsedPostFilter().namedFilters());
        }
        if (namedQueries.isEmpty()) {
            return null;
        }
        Map<String, Weight> weights = new HashMap<>();
        for (Map.Entry<String, Query> entry : namedQueries.entrySet()) {
            weights.put(
                entry.getKey(),
                context.searcher().createWeight(context.searcher().rewrite(entry.getValue()), ScoreMode.COMPLETE, 1)
            );
        }`

Reply:
i have added a potential fix, please have a look, however i might suggest that this phase is only executed if there are named queries, so i am not sure how niche the feature usage is but just FYI, might not have that high impact in case if its slow

Reply:
@msfroh @reta , if you can please have a look